### ISSUE TITLE
[Feature Request] Upload Attachments to Support Tickets

### ISSUE THREAD

**Issue by DataLores** [FOSSBilling/FOSSBilling#1409]
**Is your feature request related to a problem? Please describe.**
Customers find it very frustrating that they cannot share screenshots of problems when providing diagnostic information on support cases.

**Describe the solution you'd like**
Support an upload feature on support case tickets with limited file types and sizes.

**Describe alternatives you've considered**
None.

**Additional context**
Am not aware of any existing tickets related to the feature request.

**Comment by BelleNottelling** [comment]
Yeah, this has crossed my mind a handful of times.
I'd like us to have a upload "component" that can be used in various places of FOSSBilling and have it be able to define allowed file types, sizes, and potentially even provide some kind of interface to link it into an AV scanner if someone was includes to do so.

**Comment by John-S4** [comment]
> I'd like us to have a upload "component" that can be used in various places of FOSSBilling and have it be able to define allowed file types, sizes, and potentially even provide some kind of interface to link it into an AV scanner if someone wanted to do so.

At the risk of being over simplistic, wouldn't an easier solution be to just use ckeditor on the client support ticket side as well as the admin side, and then to enable file uploads in the editor, it gives control over accepted file types and max upload sizes via the config file. I don't know (but think it unlikely) that it has an av scanner interface but fulfils the rest of the criteria.

**Comment by BelleNottelling** [comment]
> At the risk of being over simplistic, wouldn't an easier solution be to just use ckeditor on the client support ticket side as well as the admin side, and then to enable file uploads in the editor, it gives control over accepted file types and max upload sizes via the config file. I don't know (but think it unlikely) that it has an av scanner interface but fulfils the rest of the criteria.

CKEditor is exclusively client-side, we still need to implement behavior on the back-end which facilitates and handles those file uploads

**Comment by John-S4** [comment]
> CKEditor is exclusively client-side, we still need to implement behavior on the back-end which facilitates and handles those file uploads

Yep, all true, but it's also probably not the best  front-end UI experience, uploads should be more like attachments than something inserted into the flow of text. 

This is why you I shouldn't be skimming through GH issues when I an supposed to be asleep.

**Comment by admdly** [comment]
Adding file uploads should be fairly trivial once the `Request` changes are merged, though as @BelleNottelling mentioned back in 2023 perhaps this is something we want to have in say a `Tools` function? That way we have control over things like allowed file types, paths, etc. rather than each module doing whatever it likes.

**Comment by BelleNottelling** [comment]
I was thinking probably more integrated with the system so the admins would have greater control over them
